d2common/d2fileformats/d2tbl/text_dictionary.go

```diff
@@ -89,64 +102,88 @@ func (td TextDictionary) loadHashEntry(idx int, hashEntry *textDictionaryHashEnt
 		key += string(b)
 	}
 
-	if key == "x" || key == "X" {
+	hashEntry.key = key
+
+	if hashEntry.key == "x" || hashEntry.key == "X" {
 		key = "#" + strconv.Itoa(idx)
 	}
 
-	_, exists := td[key]
+	_, exists := td.Entries[key]
 	if !exists {
-		td[key] = value
+		td.Entries[key] = hashEntry.name
 	}
 
 	return nil
 }
 
 type textDictionaryHashEntry struct {
-	IsActive    bool
-	Index       uint16
+	name        string
+	key         string
 	HashValue   uint32
 	IndexString uint32
 	NameString  uint32
+	Index       uint16
 	NameLength  uint16
+	active      byte
+}
+
+func (t *textDictionaryHashEntry) IsActive() bool {
+	return t.active > 0
 }
 
 const (
 	crcByteCount = 2
 )
 
 // LoadTextDictionary loads the text dictionary from the given data
-func LoadTextDictionary(dictionaryData []byte) (TextDictionary, error) {
-	lookupTable := make(TextDictionary)
+func LoadTextDictionary(dictionaryData []byte) (*TextDictionary, error) {
+	var err error
+
+	lookupTable := &TextDictionary{
+		Entries: make(map[string]string),
+	}
 
 	br := d2datautils.CreateStreamReader(dictionaryData)
 
 	// skip past the CRC
-	_, _ = br.ReadBytes(crcByteCount)
-
-	var err error
+	lookupTable.crcBytes, err = br.ReadBytes(crcByteCount)
```

Comment:
we may want to validate the CRC during decode if we go to the trouble of adding the CRC during encode